source/backend/tests/unit/api/Services/AcquisitionServiceTest.cs

```diff
@@ -51,42 +62,30 @@ public void Add_Success()
         public void Add_NoPermission()
         {
             // Arrange
-            var helper = new TestHelper();
-            var user = PrincipalHelper.CreateForPermission();
-            var service = helper.Create<AcquisitionFileService>(user);
+            var service = CreateAcquisitionServiceWithPermissions();
 
             var acqFile = EntityHelper.CreateAcquisitionFile();
 
-            var repository = helper.GetService<Mock<IAcquisitionFileRepository>>();
-            repository.Setup(x => x.Add(It.IsAny<PimsAcquisitionFile>())).Returns(acqFile);
-
             // Act
             Action act = () => service.Add(acqFile);
 
             // Assert
             act.Should().Throw<NotAuthorizedException>();
-            repository.Verify(x => x.Add(It.IsAny<PimsAcquisitionFile>()), Times.Never);
         }
```

Comment:
I see now what the new pattern is regarding not mocking repos when exceptions are expected to be thrown but I believe there is a subtle case of a bug that would be not longer covered by this check if the service code throws **after** calling the repo. This could happen due to developer error.

Say, this contrived example:

```cs
// poorly written service code
public PimsEntity AddWithBug(entity)
{
   var added = _repo.Add(entity);
   User.ThrowIfNotAuthorized(Permissions.SystemAdmin);
}
```

In that case the tests above in the new pattern would all pass because the service eventually throws but wouldn't we want to also make sure that no repo is called as well?

Reply:
By not checking for repo calls we are implicitly trusting the developers to write the auth check (or any other check that results in an exception) before any calls to the repos

Reply:
This is fair, there is some value to having the non-call check in the case of writes, I'll update this to keep those checks for writes but leave them out for reads as that will always be redundant.